bundlewrap/items/files.py

```diff
@@ -201,13 +272,19 @@ def content(self):
 
     @cached_property
     def content_hash(self):
-        if self.attributes['content_type'] == 'binary':
+        if self.attributes['content_type'] in ('binary', 'download'):
             return hash_local_file(self.template)
         else:
             return sha1(self.content)
 
     @cached_property
     def template(self):
+        if self.attributes['content_type'] == 'download':
+            file_path, remove_dir = download_file(self)
+            if remove_dir:
+                io.debug(_("registering {} for deletion on exit").format(remove_dir))
+                at_exit(rmtree, remove_dir, ignore_errors=True)
```

Comment:
Ugh, this is tricky. Initially I was wondering why you're using your own filesystem locking instead of Python locks. Then I thought "neat, that way other bw processes can share the cache". But with `at_exit` you might then delete stuff those other processes are still using. Off the top of my head there is no reasonable way to fix this and we should make the download cache per-process (and thus don't need the fs locking anymore).

Reply:
Nevermind.